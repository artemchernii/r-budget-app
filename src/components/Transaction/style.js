import styled from "styled-components";


const TransactionWrapper = styled.div`
    background-color: ${({value}) => value < 0 ? '#e2e1e1' : '#fff'};
    border: 1px solid #dbdbdb;
    border-radius: 3px;
    padding: 5px;
    margin-bottom: 10px;
    display: flex;
    align-items: center;
    justify-content: center;
    color: #000;
`
TransactionWrapper.displayName = 'TransactionWrapper';

const TransactionDate = styled.div`
    flex-grow: 1;
`
TransactionDate.displayName = 'Date'
const TransactionValue = styled.div`
    flex-grow: 1;
    `
TransactionValue.displayName = 'Value'
const TransactionComment = styled.div`
    flex-grow: 2;
    `
TransactionComment.displayName = 'CoTransactionComment'

export {TransactionWrapper, TransactionDate, TransactionValue, TransactionComment}