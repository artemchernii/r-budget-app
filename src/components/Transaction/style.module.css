
.transaction {
    border: 1px solid #fff;
    border-radius: 15px;
    margin: 5px 0;
    font-weight: bold;
    padding: 5px 50px;
  }
.transaction span {
    font-style: italic;
}