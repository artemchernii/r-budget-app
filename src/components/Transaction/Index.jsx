/* eslint-disable react/prop-types */
import propTypes from 'prop-types';
import style from './style.module.css';
import {
    TransactionWrapper,
    TransactionDate,
    TransactionValue,
    TransactionComment,
    Icon,
} from './style';
import { FormattedMessage } from 'react-intl';
import { useContext } from 'react';
import { stateContext } from '../../providers/context/defaultContext';
import star from '../../assets/star.svg';
import lightStar from '../../assets/light-star.svg';
import { Button } from '../App/style';
export default function Transaction({
    value,
    date,
    comment,
    id,
    onDelete,
    isFavoured,
    onFavouredClick,
}) {
    const { currency } = useContext(stateContext).state;

    const {
        state: { theme },
    } = useContext(stateContext);
    const handleDelete = () => {
        onDelete(id);
    };
    return (
        <TransactionWrapper
            data-testid="transaction"
            value={value}
            className={style.transaction}
        >
            <Icon currenttheme={theme} onClick={() => onFavouredClick(id)}>
                <img src={isFavoured ? lightStar : star} alt="star" />
            </Icon>
            <TransactionValue data-testid="Value">
                {value.toFixed(2)}{' '}
                {currency === 'UAH' ? '₴' : currency === 'USD' ? '$' : '€'}
            </TransactionValue>
            <TransactionDate>{date}</TransactionDate>
            <TransactionComment>{comment}</TransactionComment>
            <Button onClick={handleDelete}>
                <FormattedMessage id="button.delete" />
            </Button>
        </TransactionWrapper>
    );
}

Transaction.propTypes = {
    label: propTypes.string,
    value: propTypes.number,
    date: propTypes.string,
    comment: propTypes.string,
    id: propTypes.number,
    onDelete: propTypes.func,
    onFavouredClick: propTypes.func,
};
Transaction.defaultProps = {
    label: '',
    value: 0,
    date: null,
};
