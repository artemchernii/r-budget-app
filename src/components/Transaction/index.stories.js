import Transaction from "./Index";

export default {
  title: 'Example/Transaction',
  component: Transaction,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
    layout: 'centered',
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
  tags: ['autodocs'],
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    label: { control: 'radio', options: ['Increase by', 'Decrease by'] },
    value: {control: { type: 'range', min: 0, max: 300, step: 10 }}
  },
};

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Increased = {
  args: {
    transaction: {
        label: 'Increased by',
        value: 100
    }
  },
};

export const Decreased = {
  args: {
    transaction: {
        label: 'Decreased by',
        value: -50
    }
  },
};


