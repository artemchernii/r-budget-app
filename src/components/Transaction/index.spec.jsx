import { beforeEach, describe, expect, it } from 'vitest';
import Transaction from './Index/';
import { render, screen } from '@testing-library/react';

describe('Transaction component', () => {
    let sut;
    let props;

    beforeEach(() => {
        props = {
            value: 23,
            comment: 'test comment',
            date: '01.01.2002',
        };
        sut = render(<Transaction {...props} />).findBy;
    });
    it('should match transaction', () => {
        expect(sut).toMatchSnapshot();
    });

    it('should show 2 zeros after amount', () => {
        const value = screen.getByTestId('Value').textContent;
        console.log('value', value);
        expect(value).toBe('23.00');
    });
});
