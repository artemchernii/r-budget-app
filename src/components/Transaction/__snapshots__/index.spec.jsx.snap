// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Transaction component > should match transaction 1`] = `undefined`;
